File: Tree/GeneralTree/GeneralTree.java
import java.util.ArrayList;
import java.util.List;

public class GeneralTree {
    private static class TreeNode {
        String value;
        List<TreeNode> children;

        TreeNode(String value) {
            this.value = value;
            this.children = new ArrayList<>();
        }
    }

    private TreeNode root;

    public GeneralTree(String rootValue) {
        this.root = new TreeNode(rootValue);
    }

    private TreeNode search(TreeNode current, String value) {
        if (current == null) return null;
        if (current.value.equals(value)) return current;

        for (TreeNode child : current.children) {
            TreeNode found = search(child, value);
            if (found != null) return found;
        }
        return null;
    }

    public void addChild(String parentValue, String childValue) {
        TreeNode parent = search(root, parentValue);
        if (parent != null) {
            parent.children.add(new TreeNode(childValue));
        } else {
            System.out.println("Parent '" + parentValue + "' not found!");
        }
    }

    public void print(TreeNode node, int level) {
        if (node == null) return;
        
        for (int i = 0; i < level; i++) System.out.print("  ");
        System.out.println("|-- " + node.value);
        
        for (TreeNode child : node.children) {
            print(child, level + 1);
        }
    }

    public void display() {
        print(root, 0);
    }

    public static void main(String[] args) {
        GeneralTree tree = new GeneralTree("Root");
        tree.addChild("Root", "Folder_A");
        tree.addChild("Root", "Folder_B");
        tree.addChild("Folder_A", "File_A1");
        tree.addChild("Folder_A", "File_A2");
        tree.addChild("Folder_B", "File_B1");

        tree.display();
    }
}

File: Tree/GeneralTree/GeneralTree.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.children = []

class GeneralTree:
    def __init__(self, root_value):
        self.root = TreeNode(root_value)

    def search_node(self, current_node, value):
        if current_node is None:
            return None
        if current_node.value == value:
            return current_node
        

        for child in current_node.children:
            found = self.search_node(child, value)
            if found:
                return found
        return None

    def add_child(self, parent_value, child_value):
        parent_node = self.search_node(self.root, parent_value)
        if parent_node:
            new_child = TreeNode(child_value)
            parent_node.children.append(new_child)
            return True
        print(f"Parent '{parent_value}' not found!")
        return False

    def print_tree(self, node=None, level=0):
        if node is None and level == 0:
            node = self.root
            
        print("  " * level + "|-- " + str(node.value))
        for child in node.children:
            self.print_tree(child, level + 1)

if __name__ == "__main__":
    tree = GeneralTree("Root")
    tree.add_child("Root", "Folder_A")
    tree.add_child("Root", "Folder_B")
    tree.add_child("Folder_A", "File_A1")
    tree.add_child("Folder_A", "File_A2")
    tree.add_child("Folder_B", "File_B1")
    
    tree.print_tree()
